src/software/world/timestamped_possession_state.h

```diff
@@ -0,0 +1,106 @@
+#pragma once
+
+#include <map>
+#include <optional>
+
+#include "software/time/timestamp.h"
+#include "software/world/robot_state.h"
+#include "software/world/team_types.h"
+
+/**
+ * A light structure for a robot ID with TeamSide
+ */
+struct RobotIdWithTeamSide
+{
+    unsigned int id;
+    TeamSide team_side;
+
+    bool operator==(const RobotIdWithTeamSide &other) const
+    {
+        return id == other.id && team_side == other.team_side;
+    }
+
+    bool operator<(const RobotIdWithTeamSide &other) const
+    {
+        return id < other.id || team_side < other.team_side;
+    }
+};
+
+class TimestampedPossessionState
+{
+   public:
+    /**
+     * Creates a new possession state that tracks no robots' possession and has timestamp
+     * of 0
+     */
+    TimestampedPossessionState();
+
+    /**
+     * Updates the possession state
+     *
+     * @throws std::invalid_argument if update_time is older than timestamp()
+     *
+     * @param possessions The robots that have possession of the ball at update_time
+     * @param update_time The timestamp to update possession state
+     */
+    void updateState(std::vector<RobotIdWithTeamSide> possessions, Timestamp update_time);
+
+    /**
+     * Returns the timestamp of when the possession state was last updated
+     *
+     * @return the timestamp
+     */
+    Timestamp timestamp() const;
```

Comment:
perhaps rename to `lastUpdateTimestamp` just be a bit more specific?

Reply:
Apologies, this should be `getLastUpdateTimestamp` (https://github.com/UBC-Thunderbots/Software/blob/master/docs/code-style-guide.md#getter-and-setter-functions)